apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/block-iframe-preview.js

```diff
@@ -121,6 +142,24 @@ const BlockFramePreview = ( {
 		} );
 	}, [ viewportWidth ] );
 
+	// Set template title.
+	useEffect( () => {
+		const iframeBody = get( iframeRef, [ 'current', 'contentDocument', 'body' ] );
+		if ( ! iframeBody ) {
+			return;
+		}
+
+		/*
+		 * Set the template title.
+		 * The setTimeout() hack is needed for the first-rendering cycle,
+		 * since there is a race condition.
+		 * The next ones the title is updated without the delay.
+		 */
+		if ( ! setTemplateTitle( title, iframeBody ) ) {
+			setTimeout( () => setTemplateTitle( title, iframeBody ), 0 );
+		}
+	}, [ recomputeBlockListKey, iframeRef, title ] );
```

Comment:
Does `useEffect` need to recompute after all of these (`[ recomputeBlockListKey, iframeRef, title ]`), or just after `title`?